cfme/tests/cloud/test_cloud_events.py

```diff
@@ -60,24 +62,23 @@ def rm_cmp(_, y):
     provider.mgmt.remove_netsec_group(nsg_name, resource_group)
 
 
-def test_vm_capture(appliance, request, provider, setup_provider, register_event):
+def test_vm_capture(appliance, request, provider, register_event):
     """
     tests that generalize and capture vm azure events are received and parsed by CFME
 
     Metadata:
         test_flag: events, provision
     """
 
-    mgmt = provider.mgmt
     vm = appliance.collections.cloud_instances.instantiate(
         random_vm_name(context='capture'), provider)
 
-    if not mgmt.does_vm_exist(vm.name):
+    if not vm.mgmt.exists:
         vm.create_on_provider(find_in_cfme=True, allow_skip="default")
         vm.refresh_relationships()
 
     # # deferred delete vm
-    request.addfinalizer(vm.delete_from_provider)
+    request.addfinalizer(vm.mgmt.cleanup)
```

Comment:
Should use vm.cleanup_on_provider here to avoid exceptions when vm does not exist